# Who is struggling and how long will it take to bounce back
#### Not looking at defense or base running metrics

In [32]:
import pandas as pd
from pybaseball import batting_stats, pitching_stats

In [57]:
Prospects = pd.read_excel(r"C:\Users\Owner\OneDrive\Desktop\exploratory data files\Notable Top Prospects.xlsx")

In [58]:
P_2021 = list(Prospects["yr_2021"])
P_2021

['Sam Huff',
 'Michael Harris',
 'Heliot Ramos',
 'Josh Lowe',
 'JJ Bleday',
 'Colin Pache',
 'Nolan Jones',
 'MJ Melendez',
 'Tyler Freeman',
 'Mackenzie Gore',
 'Oneil Cruz',
 'Matthew Liberatore',
 'Sixto Sanchez',
 'Alek Thomas',
 'George Kirby',
 'Nick Lodolo',
 'Hunter Greene',
 'Jarren Duran',
 'Nolan Gorman',
 'Reid Detmers',
 'Vidal Brujan',
 'Shane Baz',
 'Joey Bart',
 'Riley Greene',
 'C.J. Abrams',
 'Spencer Torkelson',
 'Bobby Witt Jr.',
 'Julio Rodriguez',
 'Adley Rutschman',
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan,
 nan]

In [35]:
Today_bat = batting_stats(2022, qual = 100)

In [36]:
Prospect_bats_2021 = Today_bat.query('Name in @P_2021')[["Name", "AB","AVG", "OBP", "OPS", "wOBA", "BABIP", "SwStr%", "HR", "K%", "BB%", "wRC+", "ISO" ]]
Prospect_bats_2021                                      

Name   AB    AVG    OBP    OPS   wOBA  BABIP  SwStr%  HR  \
55     Julio Rodriguez  365  0.271  0.334  0.816  0.351  0.339   0.160  18   
58     Adley Rutschman  220  0.250  0.364  0.791  0.350  0.291   0.064   5   
160     Bobby Witt Jr.  394  0.256  0.300  0.744  0.320  0.302   0.118  15   
216        Alek Thomas  277  0.249  0.306  0.696  0.306  0.284   0.112   8   
97        Nolan Gorman  211  0.242  0.316  0.781  0.339  0.304   0.159  13   
199          Joey Bart  168  0.214  0.305  0.698  0.310  0.325   0.173   9   
117        MJ Melendez  291  0.237  0.321  0.765  0.331  0.281   0.118  14   
312          Josh Lowe  181  0.221  0.284  0.627  0.276  0.336   0.137   2   
242       Jarren Duran  171  0.234  0.288  0.680  0.297  0.319   0.114   3   
266         Oneil Cruz  158  0.209  0.254  0.678  0.289  0.267   0.151   9   
258       Riley Greene  194  0.242  0.305  0.661  0.293  0.338   0.106   3   
401       Vidal Brujan  138  0.167  0.230  0.477  0.216  0.206   0.092   2   
342  Spencer Torkelson  264  0.197  0.282  0.577  0.262  0.254   0.106   5   

        K%    BB%  wRC+    ISO  
55   0.272  0.070   135  0.211  
58   0.169  0.138   128  0.177  
160  0.229  0.052   107  0.188  
216  0.183  0.066    93  0.141  
97   0.312  0.090   121  0.223  
199  0.400  0.100   101  0.179  
117  0.255  0.114   114  0.206  
312  0.333  0.076    83  0.122  
242  0.283  0.060    88  0.158  
266  0.355  0.059    85  0.215  
258  0.291  0.080    89  0.113  
401  0.237  0.079    41  0.080  
342  0.255  0.097    68  0.098

#### Only 2021 prospects with at least 100 AB
### Most struggling in BA with some decent OPS those above .7. So you can see the power is there mostly for Julio, Bobby, Gorman, Melendez
### Most of the struggle is getting on base and plate discipline K% for the struggling hitters is and walk rates is low
## To conclude theres only a few players here I would say this far have been doing good and the rest aren't showing any signs of improvement from the plate discipline and getting on base perspective

### Some struggling players that the advanced metrics look like they could improve: Josh Lowe, Riley Greene, Joey Bart especially looking at wOBA, BABIP. Although these players still struggling with power metrics
### All these players need to focus on plate discipline which will come in time with experience hopefully
### Also, some of them look like struggling player may just turn out to be serviceable non all star players
#### Very poor performers may consider trading: torkelson, brujan. Duran may be a servicable speedster

In [45]:
Today_pit = pitching_stats(2022, qual = 5)
#list(Today_pit.columns)

In [46]:
Prospect_pitch_2021 = Today_pit.query('Name in @P_2021')[['Name', 'GS', 'IP', 'WHIP', 'ERA', 'K/9', 'BB/9', 'HR/9', 'AVG', 'Contact%', 'SwStr%', 'SIERA', 'Hard%', 'Soft%', 'HardHit%', 'CSW%', 'CStr%', 'FIP']]
Prospect_pitch_2021 

Name  GS     IP  WHIP   ERA    K/9  BB/9  HR/9    AVG  \
264        George Kirby  15   79.1  1.17  3.40   9.19  1.25  1.36  0.257   
269        Reid Detmers  17   89.0  1.04  3.44   8.29  3.13  1.21  0.195   
536       Hunter Greene  20  102.2  1.30  5.26  11.13  3.59  2.02  0.234   
479         Nick Lodolo  10   47.2  1.66  4.72  12.08  4.53  1.32  0.288   
507           Shane Baz   6   27.0  1.33  5.00  10.00  3.00  1.67  0.252   
542  Matthew Liberatore   6   27.0  1.74  5.33   7.33  4.67  1.67  0.292   

     Contact%  SwStr%  SIERA  Hard%  Soft%  HardHit%   CSW%  CStr%   FIP  
264     0.797   0.106   3.32  0.345  0.160     0.395  0.267  0.161  3.61  
269     0.782   0.103   4.09  0.326  0.178     0.407  0.265  0.162  4.25  
536     0.720   0.135   3.63  0.331  0.135     0.402  0.279  0.144  4.96  
479     0.741   0.113   3.67  0.276  0.197     0.339  0.289  0.176  4.42  
507     0.718   0.133   3.59  0.429  0.130     0.506  0.293  0.160  4.42  
542     0.823   0.078   5.09  0.374  0.110     0.495  0.235  0.156  5.46

### George Kirby and Detmers seem to be only ones performing very well
### Baz Lodolo and green seem to be good K pitchers but are still struggling a bit with hits and walk rates
### Lodolos contact let up is good so his advanced metrics look good so it may come with time with him
### The best and most consistent pitcher so far is george Kirby
#### Liberatore and baz have been set back with injuries and lodolo also has been limited with IP. So the population is a little low on them so far so they may become better with time

#### Kirby- improve contact and apposing BA and opposing contact type
#### Detmers - improve contact type maybe improve Ks
#### Greene - ERA improve soft% and hardhit% Cstr%
### Low population on rest of them

In [59]:
P_2017 = list(Prospects["yr_2017"])
P_2017

['Andrew Benintendi',
 'Yoan Moncada',
 'Gleyber Torres',
 'Dansby Swanson',
 'Amed Rosario',
 'Alex Reyes',
 'J.P. Crawford',
 'Tyler Glasnow',
 'Victor Robles',
 'Austin Meadows',
 'Ozzie Albies',
 'Lucas Giolito',
 'Cody Bellinger',
 'Eloy Jimenez',
 'Rendan Rodgers',
 'Michael Kopech',
 'Rafael Devers',
 'Lewis Brinson',
 'Mickey Moniak',
 'Francis Martes',
 'Willy Adames',
 'Bradley Zimmer',
 'Manuel Margot',
 'Jackson Frazier',
 'Anderson Espinoza',
 'Nick Senzel',
 'Josh Bell',
 'Ian Happ',
 'Kyle Lewis',
 'Corey Ray',
 'Brent Honeywell Jr.',
 'Kevin Maitan',
 'Jose DeLeon',
 'Vladimir Guerrero Jr.',
 'Kyle Tucker',
 'Tyler Oneil',
 'Black Rutherford',
 'Josh Hader',
 'Carson Kelly',
 'Francisco Mejia',
 'Jay Groome',
 'Hunter Renfroe',
 'Braxton Garrett',
 'Jeff Hoffman',
 'Reynaldo Lopez',
 'Aaron Judge',
 'Jorge Mateo',
 'Mitch Keller',
 'Yadier Alvarez',
 'Nick Gordon',
 'Riley Pint',
 'Franklin Barreto',
 'Kolby Allard ',
 'David Paulino',
 'Leody Taveras',
 'Yohander Mende

In [60]:
hit_2017 = batting_stats(2017, qual = 100).query('Name in @P_2017')[["Name", "AB","AVG", "OBP", "OPS", "wOBA", "BABIP", "SwStr%", "HR", "K%", "BB%", "wRC+", "ISO" ]]
hit_2017

Name   AB    AVG    OBP    OPS   wOBA  BABIP  SwStr%  HR  \
2          Aaron Judge  542  0.284  0.422  1.049  0.430  0.357   0.133  52   
30      Cody Bellinger  480  0.267  0.352  0.933  0.380  0.299   0.135  39   
158       Matt Chapman  290  0.234  0.313  0.785  0.332  0.290   0.115  14   
268      Manuel Margot  487  0.263  0.313  0.721  0.309  0.309   0.086  13   
95            Ian Happ  364  0.253  0.328  0.842  0.349  0.316   0.161  24   
101       Ozzie Albies  217  0.286  0.354  0.810  0.347  0.316   0.103   6   
154  Andrew Benintendi  573  0.271  0.352  0.776  0.332  0.301   0.077  20   
307     Bradley Zimmer  299  0.241  0.307  0.692  0.299  0.328   0.139   8   
178       Yoan Moncada  199  0.231  0.338  0.750  0.327  0.325   0.126   8   
28        Jesse Winker  121  0.298  0.375  0.904  0.384  0.322   0.068   7   
109      Rafael Devers  222  0.284  0.338  0.819  0.344  0.342   0.126  10   
233     Hunter Renfroe  445  0.231  0.284  0.751  0.316  0.275   0.150  26   
131          Josh Bell  549  0.255  0.334  0.800  0.338  0.278   0.089  26   
122  Jeimer Candelario  127  0.283  0.359  0.784  0.342  0.351   0.125   3   
44        Jorge Alfaro  107  0.318  0.360  0.874  0.369  0.420   0.217   5   
332       Derek Fisher  146  0.212  0.307  0.663  0.291  0.299   0.114   5   
204       Raimel Tapia  160  0.288  0.329  0.754  0.322  0.361   0.119   2   
372     Dansby Swanson  488  0.232  0.312  0.636  0.276  0.292   0.096   6   
353       Amed Rosario  165  0.248  0.271  0.665  0.282  0.330   0.181   4   
422   Christian Arroyo  125  0.192  0.244  0.548  0.237  0.231   0.092   3   

        K%    BB%  wRC+    ISO  
2    0.307  0.187   174  0.343  
30   0.266  0.117   138  0.315  
158  0.282  0.098   110  0.238  
268  0.200  0.066    90  0.146  
95   0.312  0.094   114  0.261  
101  0.148  0.086   110  0.171  
154  0.170  0.106   102  0.154  
307  0.298  0.078    79  0.144  
178  0.320  0.126   105  0.181  
28   0.175  0.109   134  0.231  
109  0.238  0.075   110  0.198  
233  0.292  0.056    94  0.236  
131  0.189  0.106   108  0.211  
122  0.211  0.092   112  0.142  
44   0.289  0.026   126  0.196  
332  0.325  0.102    82  0.144  
204  0.211  0.047    82  0.138  
372  0.218  0.107    63  0.092  
353  0.288  0.018    75  0.145  
422  0.237  0.059    44  0.112

### first year strugglers - zimmer, moncada, swanson, rosario, arroyo, fisher, renfroe, chapman - BA
### Power struggles - rosario, swanson, devers, moncada, zimmer, albies, margot, chapman, winker, candelario, alfara, fisher, tapia, arroyo (less than 20 hr) 
### OPS struggles - margot, benny, zimmer, moncada, bell, swanson, rosario, chapman, renfroe, candelario, fisher, tapia, arroyo - <.8 ops
### advanced on base struggles - margot, swanson, zimmer, rosario, renfroe, fisher, arroyo, bellinger, chapman, bell
### K% - bellinger, happ, zimmer, moncada, candelario, alfaro, fisher, tapia, arroyo, devers, swanon, judge, chapman, rosario >20%
### B% - margot, zimmer, devers, arroyo, rosario, tapia, alfaro, renfroe

## Biggest strugglers - zimmer, moncada, Amed rosario, swanson, margot, chapman, renfroe, fisher, arroyo, candelario, tapia

In [64]:
pit_2017 = pitching_stats(2017, qual = 5).query('Name in @P_2017')[['Name', 'GS', 'G', 'IP', 'WHIP', 'ERA', 'K/9', 'BB/9', 'HR/9', 'AVG', 'Contact%', 'SwStr%', 'SIERA', 'Hard%', 'Soft%', 'HardHit%', 'CSW%', 'CStr%', 'FIP']]
pit_2017

Name  GS   G     IP  WHIP   ERA    K/9  BB/9  HR/9    AVG  \
332     German Marquez  29  29  162.0  1.38  4.39   8.17  2.72  1.39  0.270   
233        Luke Weaver  10  13   60.1  1.26  3.88  10.74  2.54  1.04  0.252   
522       Jeff Hoffman  16  23   99.1  1.47  5.89   7.43  3.62  1.36  0.268   
313       Sean Newcomb  19  19  100.0  1.57  4.32   9.72  5.13  0.90  0.254   
31          Josh Hader   0  35   47.2  0.99  2.08  12.84  4.15  0.76  0.154   
393     Reynaldo Lopez   8   8   47.2  1.32  4.72   5.66  2.64  1.32  0.255   
378  Robert Stephenson  11  25   84.2  1.58  4.68   9.14  5.63  1.28  0.247   
42       Lucas Giolito   7   7   45.1  0.95  2.38   6.75  2.38  1.59  0.189   
518     Francis Martes   4  32   54.1  1.51  5.80  11.43  5.13  1.16  0.243   
560      David Paulino   6   6   29.0  1.48  6.52  10.55  2.17  2.48  0.298   
227      Carson Fulmer   5   7   23.1  1.24  3.86   7.33  5.01  1.54  0.186   
611     Walker Buehler   0   8    9.1  2.04  7.71  11.57  7.71  1.93  0.306   
442    Yohander Mendez   0   7   12.1  1.30  5.11   5.11  2.19  2.19  0.271   
647        Erick Fedde   3   3   15.1  2.15  9.39   8.80  4.70  2.93  0.373   
607      Tyler Glasnow  13  15   62.0  2.02  7.69   8.13  6.39  1.89  0.313   
597       Amir Garrett  14  16   70.2  1.61  7.39   8.02  5.09  2.93  0.265   

     Contact%  SwStr%  SIERA  Hard%  Soft%  HardHit%   CSW%  CStr%   FIP  
332     0.811   0.091   4.27  0.345  0.177     0.362  0.261  0.170  4.40  
233     0.792   0.097   3.29  0.259  0.191     0.340  0.287  0.190  3.17  
522     0.817   0.083   4.88  0.344  0.201     0.360  0.262  0.180  4.80  
313     0.759   0.112   4.71  0.270  0.175     0.344  0.276  0.164  4.19  
31      0.653   0.171   2.99  0.319  0.191     0.255  0.323  0.153  3.03  
393     0.820   0.090   5.44  0.278  0.179     0.272  0.232  0.142  4.75  
378     0.723   0.126   4.99  0.384  0.128     0.372  0.266  0.140  4.92  
42      0.795   0.101   4.49  0.354  0.192     0.338  0.240  0.139  4.94  
518     0.717   0.126   4.02  0.348  0.199     0.376  0.297  0.171  4.45  
560     0.762   0.106   3.75  0.405  0.107     0.414  0.280  0.174  5.12  
227     0.821   0.087   5.59  0.209  0.194     0.299  0.222  0.136  5.69  
611     0.762   0.104   4.47  0.208  0.208     0.333  0.286  0.182  5.94  
442     0.833   0.084   5.00  0.317  0.195     0.390  0.209  0.126  6.16  
647     0.864   0.057   4.45  0.346  0.154     0.346  0.262  0.205  7.20  
607     0.802   0.083   5.62  0.315  0.202     0.355  0.257  0.174  6.30  
597     0.809   0.087   5.19  0.407  0.134     0.403  0.244  0.158  7.39

### Almost everyone struggled on AVG, WHIP or FIP first year
### ERA - marquez, hoffman, reynaldo, stephenson, martes, paulino, buehler, mendez, fedde, glasnow, garrett
### K/9 - hoffman, fulmer, Reynaldo, giolito, mendez
### BB - hoffman, newcomb, hader, stephenson, martes, fulmer, buehler, fedde, glasnow, garrett
### CSW - marquez, hoffman, lopez, giolito, fulmer, mendez, glasnow, garrett, fedde, stephenson
### SIERRA - marquez, hoffman, newcomb, lopez, stephenson, giolito, martes, fulmer, buehler, mendez, fedde, glasnow, garrett

### There are other metrics but it was similar trends
### hoffman, reynaldo, garrett, glasnow, fedde, buehler, martes, stephenson, fulmer, marquez, giolito, mendez
#### most of pitchers struggled

In [65]:
Strug_pitch = ["Jeff Hoffman", "Reynaldo Lopez", "Amir Garrett", "Tyler Glasnow", "Erick Fedde", "Francis Martes", "Robert Stephenson", "Carson Fulmer", "Walker Buehler", "German Marquez", "Lucas Giolito", "Yohander Mendez"]
Strug_hit = ['Bradley Zimmer', 'Yoan Moncada', 'Amed Rosario', 'Dansby Swanson', 'Manuel Margot', 'Matt Chapman', 'Hunter Renfroe', 'Derek Fisher', 'Christian Arroyo', 'Jeimer Candelario', 'Raimel Tapia']

In [72]:
YY_2017_bat = batting_stats(2017, 2022, qual = 100).query('Name in @Strug_hit')[["Name", "G", "PA", "AB","AVG", "OBP", "OPS", "wOBA", "BABIP", "SwStr%", "HR", "K%", "BB%", "wRC+", "ISO", "Season"]]
YY_2017_bat = YY_2017_bat.sort_values(by = ["Name", "Season"], ascending = False) 
YY_2017_bat

Name    G   PA   AB    AVG    OBP    OPS   wOBA  BABIP  \
2283       Yoan Moncada   67  268  245  0.196  0.261  0.580  0.257  0.255   
538        Yoan Moncada  144  616  520  0.263  0.375  0.787  0.347  0.350   
1396       Yoan Moncada   52  231  200  0.225  0.320  0.705  0.309  0.315   
166        Yoan Moncada  132  559  511  0.315  0.367  0.915  0.379  0.406   
1356       Yoan Moncada  149  650  578  0.235  0.315  0.714  0.311  0.344   
948        Yoan Moncada   54  231  199  0.231  0.338  0.750  0.327  0.325   
1631       Raimel Tapia   89  301  286  0.276  0.297  0.689  0.300  0.325   
1514       Raimel Tapia  133  533  487  0.273  0.327  0.699  0.305  0.306   
711        Raimel Tapia   51  206  184  0.321  0.369  0.772  0.338  0.392   
1463       Raimel Tapia  138  447  426  0.275  0.309  0.724  0.307  0.341   
1085       Raimel Tapia   70  171  160  0.288  0.329  0.754  0.322  0.361   
592        Matt Chapman  104  410  363  0.242  0.322  0.796  0.344  0.286   
1349       Matt Chapman  151  622  529  0.210  0.314  0.716  0.311  0.272   
765        Matt Chapman   37  152  142  0.232  0.276  0.812  0.335  0.291   
433        Matt Chapman  156  670  583  0.249  0.342  0.848  0.354  0.270   
247        Matt Chapman  145  616  547  0.278  0.356  0.864  0.369  0.338   
844        Matt Chapman   84  326  290  0.234  0.313  0.785  0.332  0.290   
570       Manuel Margot   51  200  182  0.302  0.365  0.788  0.346  0.361   
1600      Manuel Margot  125  464  421  0.254  0.313  0.696  0.301  0.281   
1612      Manuel Margot   47  159  145  0.269  0.327  0.679  0.301  0.317   
1716      Manuel Margot  151  441  398  0.234  0.304  0.691  0.296  0.272   
1857      Manuel Margot  141  519  477  0.245  0.292  0.675  0.288  0.281   
1394      Manuel Margot  126  529  487  0.263  0.313  0.721  0.309  0.309   
2097  Jeimer Candelario   90  336  309  0.204  0.268  0.617  0.273  0.242   
587   Jeimer Candelario  149  626  557  0.271  0.351  0.795  0.344  0.333   
214   Jeimer Candelario   52  206  185  0.297  0.369  0.872  0.373  0.372   
1943  Jeimer Candelario   94  386  335  0.203  0.306  0.643  0.283  0.262   
1369  Jeimer Candelario  144  619  539  0.224  0.317  0.710  0.311  0.279   
636   Jeimer Candelario   38  142  127  0.283  0.359  0.784  0.342  0.351   
611      Hunter Renfroe   76  313  282  0.245  0.308  0.801  0.343  0.266   
594      Hunter Renfroe  144  572  521  0.259  0.315  0.816  0.344  0.284   
2031     Hunter Renfroe   42  139  122  0.156  0.252  0.645  0.277  0.141   
1133     Hunter Renfroe  140  494  440  0.216  0.289  0.778  0.320  0.239   
703      Hunter Renfroe  117  441  403  0.248  0.302  0.805  0.338  0.271   
1236     Hunter Renfroe  122  479  445  0.231  0.284  0.751  0.316  0.275   
1920       Derek Fisher   57  167  146  0.185  0.287  0.657  0.285  0.244   
1797       Derek Fisher   53  166  146  0.212  0.307  0.663  0.291  0.299   
447      Dansby Swanson  112  474  436  0.294  0.348  0.811  0.353  0.380   
1077     Dansby Swanson  160  653  588  0.248  0.311  0.760  0.322  0.297   
516      Dansby Swanson   60  264  237  0.274  0.345  0.809  0.348  0.350   
1201     Dansby Swanson  127  545  483  0.251  0.325  0.748  0.317  0.300   
1774     Dansby Swanson  136  533  478  0.238  0.304  0.699  0.293  0.290   
2056     Dansby Swanson  144  551  488  0.232  0.312  0.636  0.276  0.292   
946    Christian Arroyo   53  170  157  0.274  0.320  0.753  0.327  0.297   
832    Christian Arroyo   57  181  164  0.262  0.324  0.769  0.332  0.325   
2413   Christian Arroyo   34  135  125  0.192  0.244  0.548  0.237  0.231   
1629     Bradley Zimmer   99  348  299  0.227  0.325  0.669  0.300  0.347   
2150     Bradley Zimmer   34  114  106  0.226  0.281  0.611  0.270  0.367   
1647     Bradley Zimmer  101  332  299  0.241  0.307  0.692  0.299  0.328   
1192       Amed Rosario  105  457  432  0.285  0.319  0.729  0.318  0.324   
1248       Amed Rosario  141  588  550  0.282  0.321  0.731  0.315  0.340   
2041       Amed

### Results seem very inconsistent - sometimes players improve BA or Power after what they did rookie season, However if they had incredibly poor performance first two seasons they typically dont get resigned or become utility player after that time
### example - arroyo, fisher, zimmer
### you might get some servicable years like for moncada but first year is a platform you can base the career off of
## however overall everything is very inconsistent
### Unless your first two years are very poor players can always improve - swanson. In general though if they perform decent the first year you can assume they will at least be a servicable player like tapia
#### It might not be the type of player you would have like but they will typically become a servicable MLB player

In [73]:
YY_2017_pit = pitching_stats(2017, 2022, qual = 5).query('Name in @Strug_pitch')[['Name', 'GS', 'G', 'IP', 'WHIP', 'ERA', 'K/9', 'BB/9', 'HR/9', 'AVG', 'Contact%', 'SwStr%', 'SIERA', 'Hard%', 'Soft%', 'HardHit%', 'CSW%', 'CStr%', 'FIP', 'Season']]
YY_2017_pit = YY_2017_pit.sort_values(by = ["Name", "Season"], ascending = False) 
YY_2017_pit

Name  GS   G     IP  WHIP   ERA    K/9  BB/9  HR/9    AVG  \
2931  Yohander Mendez   5   8   27.2  1.55  5.53   5.86  4.88  1.30  0.264   
2678  Yohander Mendez   0   7   12.1  1.30  5.11   5.11  2.19  2.19  0.271   
1709   Walker Buehler  12  12   65.0  1.29  4.02   8.03  2.35  1.11  0.263   
429    Walker Buehler  33  33  207.2  0.97  2.47   9.19  2.25  0.82  0.197   
1160   Walker Buehler   8   8   36.2  0.95  3.44  10.31  2.70  1.72  0.178   
...               ...  ..  ..    ...   ...   ...    ...   ...   ...    ...   
3209     Amir Garrett   0  63   47.2  1.57  6.04  11.52  5.48  1.70  0.247   
423      Amir Garrett   0  21   18.1  0.93  2.45  12.76  3.44  1.96  0.161   
953      Amir Garrett   0  69   56.0  1.41  3.21  12.54  5.63  1.13  0.213   
1977     Amir Garrett   0  66   63.0  1.29  4.29  10.14  3.57  1.14  0.237   
3636     Amir Garrett  14  16   70.2  1.61  7.39   8.02  5.09  2.93  0.265   

      Contact%  SwStr%  SIERA  Hard%  Soft%  HardHit%   CSW%  CStr%   FIP  \
2931     0.804   0.087   5.69  0.364  0.170     0.284  0.220  0.132  5.37   
2678     0.833   0.084   5.00  0.317  0.195     0.390  0.209  0.126  6.16   
1709     0.765   0.117   3.83  0.310  0.183     0.416  0.262  0.145  3.81   
429      0.763   0.116   3.73  0.306  0.174     0.369  0.296  0.181  3.16   
1160     0.734   0.123   3.80  0.258  0.161     0.366  0.302  0.178  4.36   
...        ...     ...    ...    ...    ...       ...    ...    ...   ...   
3209     0.672   0.143   3.99  0.448  0.184     0.464  0.331  0.188  4.89   
423      0.571   0.181   2.82  0.306  0.139     0.417  0.355  0.174  4.34   
953      0.629   0.163   3.95  0.388  0.186     0.318  0.327  0.165  4.14   
1977     0.703   0.137   3.49  0.370  0.164     0.358  0.284  0.148  3.89   
3636     0.809   0.087   5.19  0.407  0.134     0.403  0.244  0.158  7.39   

      Season  
2931    2018  
2678    2017  
1709    2022  
429     2021  
1160    2020  
...      ...  
3209    2021  
423     2020  
953     2019  
1977    2018  
3636    2017  

[61 rows x 20 columns]

## Similar abnormalities and trends in pitchers

### Baseball America has an article that says that 70% of top prospects with success being at least an average WAR of 1.5. I think this shows that the players in a small sample size that this is not a good measurement of success and that more players can at least be average MLB player. It might disappoint the type of quality that you expected but they can at least be servicable starters

## I would say teams should wait 2 seasons before selling because players do not typically bounce back after that
## Some players may even improve like dansby or chapman did after their rookie year
### Keep an eye on all those mentioned before that have struggled so far and just all of them in general. All of them are struggling in some areas even Julio who is consensus AL rookie of the year has K% and BB% that are low
## Players to sell: Brujan, Duran maybe give them consistent starting time but this is their second season with poor performance
### It will be interesting to understand whether power metrics have fallen since minor league experience